### PR TITLE
NEO-1376 // add default to `aria-label`, update TS+error-message to force use of `aria-label*` (XOR)

### PR DESCRIPTION
**Pull request by joe-s-avaya**
[Link to updated components in Deploy Preview](UPDATEMEWITHALINK)

**Before tagging the team for a review, I have done the following:**
- [x] reviewed my code changes
- [x] ensured that all tests pass
- [x] updated the link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] tagged Matt if any visual changes have occurred
- [x] done an accessibility check on my work

Use "Hide whitespace" when looking at "Files changed"
Screenshot 2023-01-03 at 1 37 05 PM

As stated in the previous PR, we were already enforcing the use of `aria-label`, so this is just adding a default and telling the user that they _must_ pass one (and only one, ie, XOR) of `aria-label` or `aria-labelledby`.

### CODE REVIEW

File: src/components/LeftNav/LeftNav.test.tsx [hunk 1]
```diff
@@ -1,103 +1,140 @@
...
-    let renderResult: RenderResult<
-      typeof import("@testing-library/dom/types/queries"),
-      HTMLElement,
-      HTMLElement
-    >;
-
-    beforeEach(() => {
-      renderResult = render(<Default />);
+describe("LeftNav", () => {
+  describe("base tests", () => {
```
Comment (yangAtSpoken) [inline src/components/LeftNav/LeftNav.test.tsx:14]: Call this a11y tests, perhaps?  And add a positive test case where aria-labelledby is used.

Comment (joe-s-avaya) [reply src/components/LeftNav/LeftNav.test.tsx:14]: 🤦  good call on both accounts, will update, thanks!

File: src/components/LeftNav/LeftNav.test.tsx [hunk 2]
```diff
@@ -1,103 +1,140 @@
...
-    beforeEach(() => {
-      renderResult = render(<CategoryGroups />);
-    });
-
-    it("should render ok", () => {
-      const { container } = renderResult;
-      expect(container).not.toBe(null);
+  describe("storybook tests", () => {
+    describe("Default", () => {
+      let renderResult: RenderResult<
```
Comment (yangAtSpoken) [inline src/components/LeftNav/LeftNav.test.tsx:49]: nit: extract RenderResult type out like this:

type Queries = typeof import("@testing-library/dom/types/queries");
type RenderResultType = RenderResult<
  Queries,
  HTMLElement,
  HTMLElement
>;

And this line could just be:

    let renderResult: RenderResultType;

**Review by yangAtSpoken — APPROVED** [review]
A couple of nits.

File: src/components/LeftNav/LeftNav.test.tsx (hunk 1 above)
Comment (joe-s-avaya) [reply src/components/LeftNav/LeftNav.test.tsx]: updated!

File: src/components/LeftNav/LeftNav.test.tsx (hunk 2 above)
Comment (joe-s-avaya) [reply src/components/LeftNav/LeftNav.test.tsx]: Ooh! Yes! I was in the habit copy-pasting and didn't even think of that.

I've centralized the type and removed all copy-pasta (using find-all).

**Pull request merged**